support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -220,19 +237,40 @@ class TestUserFromRequest(identityService: IdentityService, testUsers: TestUserS
 
 }
 
-class GetCorporateCustomer(dynamoLookup: DynamoTableAsyncForUser) {
+class CodeValidator(zuoraLookupService: ZuoraGiftLookupService, dynamoLookup: DynamoTableAsyncForUser) {
+
+  def validate(inputCode: String, isTestUser: Boolean)(implicit ec: ExecutionContext): EitherT[Future, String, ReaderType] =
```

Comment:
```suggestion
  def validationErrorOrReaderType(code: String, isTestUser: Boolean)(implicit ec: ExecutionContext): EitherT[Future, String, ReaderType] =
```

Reply:
I would rather keep the method name as the verb for what is happening 'validate' and allow the return type to specify what is returned. It feels redundant to include the return type in the method name.

Reply:
fair enough.  Sounds like I am thinking of the FP way of "only the result matters" whereas you are leaning more to the "ask it to do something"